framework/test/unit/modules/interoperability/mainchain/module.spec.ts

```diff
@@ -0,0 +1,1185 @@
+												blsKey: Buffer.from(
+													'c1d3c7919a4ea7e3b5d5b0068513c2cd7fe047a632e13d9238a51fcd6a4afd7ee16906978992a702bccf1f0149fa5d39',
+													'hex',
+												),
+												bftWeight: BigInt(200),
+											},
+										],
+										// totalWeight / BigInt(3) + BigInt(1) = (100 + 200)/3 + 1 = 101
+										// totalWeight / BigInt(3) + BigInt(1) > certificateThreshold
+										certificateThreshold: BigInt(10), // 101 > 10
+									},
+								},
+							],
+						},
+						params,
+					);
+
+					await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+						`Invalid certificateThreshold input.`,
+					);
+				});
+
+				it(`should throw error if certificateThreshold > totalWeight`, async () => {
+					const context = createInitGenesisStateContext(
+						{
+							...genesisInteroperability,
+							chainInfos: [
+								{
+									...chainInfo,
+									chainValidators: {
+										activeValidators: [
+											{
+												blsKey: Buffer.from(
+													'901550cf1fde7dde29218ee82c5196754efea99813af079bb2809a7fad8a053f93726d1e61ccf427118dcc27b0c07d9a',
+													'hex',
+												),
+												bftWeight: BigInt(10),
+											},
+										],
+										certificateThreshold: BigInt(20),
+									},
+								},
+							],
+						},
+						params,
+					);
+
+					await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+						`Invalid certificateThreshold input.`,
+					);
+				});
+			});
+		});
+
+		// it is defined here, since it applies to both chainData & chainValidators
+		describe('validatorsHash', () => {
+			it(`should throw error if invalid validatorsHash provided`, async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...genesisInteroperability,
+						chainInfos: [
+							{
+								...chainInfo,
+								chainValidators: {
+									activeValidators,
+									certificateThreshold: BigInt(10),
+								},
+							},
+						],
+					},
+					params,
+				);
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					'Invalid validatorsHash from chainData.lastCertificate.',
+				);
+			});
+
+			it(`should not throw error if valid validatorsHash provided`, async () => {
+				certificateThreshold = BigInt(10);
+				const context = createInitGenesisStateContext(
+					{
+						...genesisInteroperability,
+						chainInfos: [
+							{
+								...chainInfo,
+								chainData: {
+									...chainData,
+									lastCertificate: {
+										...lastCertificate,
+										validatorsHash: computeValidatorsHash(activeValidators, certificateThreshold),
+									},
+								},
+								chainValidators: {
+									activeValidators,
+									certificateThreshold,
+								},
+							},
+						],
+					},
+					params,
+				);
+
+				await expect(interopMod.initGenesisState(context)).resolves.toBeUndefined();
+			});
+		});
+
+		describe('terminatedStateAccounts', () => {
```

Comment:
I think its missing test case added in [PR](https://github.com/LiskHQ/lisk-sdk/pull/8239) `should throw if there is no entry for terminateStateAccount chainID in chainInfos`